src/renderer/components/open-project-page/index.tsx

```diff
@@ -19,6 +20,37 @@ interface ProjectRowProps {
   selected: boolean;
 }
 
+interface OpenProjectPageProps {
+  onResume: Consumer<void>;
+  onNext: Consumer<void>;
+  onBack: MouseEventHandler<HTMLElement>;
+}
+
+const reservedStates = [
+  'AieApp$keyOctave',
+  'AieApp$projectFolder',
+  'AieApp$recordingProject',
+  'AieApp$recordingSet',
+  'AieApp$recordingList',
+  'AieApp$pageStateIndex',
+  'ConfigureRecordingSetPage$recordingSets',
+  'ConfigureRecordingSetPage$projectFile',
+  'RecordingPage$index',
+];
+
+const ResumeCheck: FC<OpenProjectPageProps> = ({ onResume }) => {
+  let reservedStateValues = [];
+  useEffectOnce(() => {
+    reservedStateValues = reservedStates.map((state) => localStorage.getItem(state));
+    if (!(reservedStateValues.indexOf(null) > -1 && reservedStateValues.indexOf(null) != reservedStates.length + 1)) {
+      if (confirm('Do you want to resume your previous records?')) {
+        onResume();
+      }
+    }
+  });
+  return null;
```

Comment:
This is not a React Functional Component - what you need is a Custom Hook, aka. just a plain function.